client.go

```diff
@@ -2079,6 +2084,48 @@ func (c *ToznySDKV3) MakeSecretResponse(secretRecord *pdsClient.Record, groupID
 	return secret
 }
 
+type ShareSecretInfo struct {
+	SecretName    string
+	SecretType    string
+	UsernameToAdd string
+}
+
+// ShareSecretWithUsername shares all versions of a specified secret with the user with UsernameToAdd
+func (c *ToznySDKV3) ShareSecretWithUsername(ctx context.Context, params ShareSecretInfo) error {
+	if len(params.UsernameToAdd) == 0 {
+		return fmt.Errorf("Username to add must be specified.")
+	}
+	// find the clientIDs for each username
+	searchParams := identityClient.SearchRealmIdentitiesRequest{
+		RealmName:         c.CurrentIdentity.Realm,
+		IdentityUsernames: []string{params.UsernameToAdd},
+	}
+	identities, err := c.E3dbIdentityClient.SearchRealmIdentities(ctx, searchParams)
```

Comment:
This endpoint doesnt error if it doesn't finding the identity, it passes back identities it did find ( dont punish the user ), so double check the value before using it on line 2113